### DISCUSSION TITLE
Cloud Shade with prescribed clouds

### DISCUSSION THREAD

**Discussion by steflhermitte** [tylermorganwall/rayshader#206] [Q&A]
Dear,

First al all I would like to thank Tyler and collaborators for the awesome package!

Second, I wondered if there is any option/plan to include prescribed cloud cover in the cloud shade function (e.g. given a spatially variable cloud cover/ cloud height). It would be excellent to visualise satellite cloud data over a given landscape.

Best regards and thanks again!
Stef Lhermitte

**Comment by tylermorganwall** [comment]
Do you have an example of the source data for the clouds?

**Comment by steflhermitte** [comment]
Dear Tyler,

I have no worked out example yet, but was thinking of experimenting with the [MODIS satellite cloud product](https://ladsweb.modaps.eosdis.nasa.gov/missions-and-measurements/products/MCD06COSP_D3_MODIS), [Sentinel-5 satellite cloud product](https://developers.google.com/earth-engine/datasets/tags/climate#bands), or any of the climate/weather model outputs (e.g. [Mesoscale Analysis](https://developers.google.com/earth-engine/datasets/catalog/NOAA_NWS_RTMA)).

In their simplest form these data sets have typically only a spatially explicit layer of cloud cover percentage/fraction instead of a fixed value in rayshader. More advanced data sets (e.g. Sentinel-5) would also have spatially explicit layers of cloud top/base height/cloud optical depth similar to the fixed rayshader values of start_altitude, end_altitude, attenuation_coef.

By looking at the source code of rayshader I could imagine replacing  the `generate_cloud_layer` function with a `read_cloud_layer` function that would the read in the spatially explicit cloud data, but I haven't found the time to develop it. The only potential bottlenecks for appearance though would be that i) the cloud layers typically only exist at coarser spatial scale than the heightmap, ii) that it would not generate a cloud looking feel (thanks to the Perlin Noise) but rather blocky cloud structure. So I am now thinking on how to combine the Perlin noise generator with spatially explicit inputs, but haven't figured that out yet. Additionally, I would need to find time to develop a `read_cloud_layer`, so no easy solution I guess (at least not from my perspective)

Thanks again and best regards,
Stef
